zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/JobStreamLifecycleIT.java

```diff
@@ -168,6 +173,53 @@ void shouldAggregateStreams() {
     }
   }
 
+  @RegressionTest("https://github.com/camunda/zeebe/issues/17513")
+  void shouldAggregateStreamsEvenAcrossRestarts() {
+    // given - many logically equivalent streams
+    final List<JobWorker> workers = new ArrayList<>();
+    for (int i = 0; i < 100; i++) {
+      workers.add(
+          client
+              .newWorker()
+              .jobType(jobType)
+              .handler((c, j) -> {})
+              .fetchVariables("foo", "bar")
+              .timeout(Duration.ofMillis(500))
+              .name("command")
+              .streamEnabled(true)
+              .open());
+    }
+    Awaitility.await("until streams are registered")
+        .untilAsserted(
+            () ->
+                JobStreamActuatorAssert.assertThat(JobStreamActuator.of(gateway))
+                    .clientStreams()
+                    .haveJobType(100, jobType));
+
+    // when - trigger stream restarts by restarting the gateway
+    gateway.stop().start();
+    CLUSTER.awaitCompleteTopology();
+    Awaitility.await("until streams are re-registered")
+        .atMost(Duration.ofSeconds(30))
+        .untilAsserted(
+            () ->
+                JobStreamActuatorAssert.assertThat(JobStreamActuator.of(gateway))
+                    .clientStreams()
+                    .haveJobType(100, jobType));
+
+    // then - only one stream is registered on each broker as it is aggregated per gateway
+    for (int nodeId = 0; nodeId < 2; nodeId++) {
+      final var actuator = brokerActuator(nodeId);
+      Awaitility.await("until stream is registered on broker '%d'".formatted(nodeId))
+          .untilAsserted(
+              () ->
+                  JobStreamActuatorAssert.assertThat(actuator)
+                      .remoteStreams()
+                      .haveConsumerCount(1, jobType, 1));
+    }
+    CloseHelper.quietCloseAll(workers);
```

Comment:
It's not really necessary - the client will be closed after the test, which closes all the workers. It's just to avoid unnecessary noise when the test ends.